website_seo/models/ir_ui_view.py

```diff
@@ -52,33 +52,28 @@ class View(models.Model):
         '"how-to-do-it-right" is rendered when calling '
         '"/ecommerce/study/how-to-do-it-right".'
     )
+    seo_url_children = fields.One2many('ir.ui.view', 'seo_url_parent', 'SEO Children')
 
-    @api.multi
-    def writeToDo(self, vals):
-        # ToDo: add automatically setup of the field seo_url_level when the
-        # seo_url_parent is changed, maybe use an onchange event instead of the
-        # write() function
-        # old_seo_url_parent = False
-        # check_views = False
-        if 'seo_url_parent' in vals:
-            # check_views = True
-            if vals['seo_url_parent']:
-                parent = self.browse(vals['seo_url_parent'])
-                vals['seo_url_level'] = parent.seo_url_level + 1
-                # old_seo_url_parent = parent
-            else:
-                # if self.seo_url_parent:
-                    # old_seo_url_parent = self.seo_url_parent
-                vals['seo_url_level'] = 0
+    @api.onchange('seo_url_parent')
+    def onchange_seo_url_parent(self):
+        url_level = 0
+        if self.seo_url_parent:
+            url_level = self.seo_url_parent.seo_url_level + 1
+        self.seo_url_level = url_level
 
+    @api.multi
+    def write(self, vals):
         res = super(View, self).write(vals)
-
-        # ToDo: update all related views with the correct seo_url_level and
-        # seo_url_parent
-        # if old_seo_url_parent:
-
+        if 'seo_url_parent' in vals or 'seo_url_level' in vals:
+            self.update_related_views()
         return res
 
+    @api.multi
+    def update_related_views(self):
```

Comment:
@kaerdsar I don't think we can go this easy way. Imagine you have the url "/parent/child/baby/egg" (it will be a rare case but it is possible). If you now change the parent of "child" the seo_url_level of "baby" will only be changed but not the seo_url_level of "egg".